lib/scarpe/line.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Scarpe
+  class Line < Scarpe::Widget
+    display_properties :left, :top, :x2, :y2
+
+    def initialize(left, top, x2, y2)
+      @left = convert_to_integer(left, "left")
+      @top = convert_to_integer(top, "top")
+      @x2 = convert_to_integer(x2, "x2")
+      @y2 = convert_to_integer(y2, "y2")
+
+      super()
+      create_display_widget
+    end
+
+    private
+
+    def convert_to_integer(value, attribute_name)
+      begin
```

Comment:
Interesting! With coordinates, it can often be useful to allow "invalid" coordinates like negative numbers so that you can do things like draw a triangle partially on- and partially off-screen. Do we know if Webview will handle it if we just let negative numbers through?

Reply:
We also may want to move methods to validate parameters somewhere like Scarpe::Widget since line isn't the only class that will want them.

Reply:
we can surely use negative numbers, ig it makes more sense to remove for left and top?